framework/src/modules/chain/blocks/blocks.js

```diff
@@ -452,23 +464,27 @@ class Blocks extends EventEmitter {
 
 	async _receiveBlockFromNetworkV2(block) {
 		// Current time since Lisk Epoch
-		// Better to do it here rather than in the Sequence so receiving time is more accurate
-		const newBlockReceivedAt = this.slots.getTime();
+		block.receivedAt = this.slots.getSlotTime(this.slots.getSlotNumber());
```

Comment:
We want to check if the block was received within a slot time window. If the block was received in the next slot the time would've been set to the first second of the next slot and the condition would be evaluated correctly.

Reply:
Setting the receivedAt time to the slot beginning time does not add any value. However, it adds a little bit of obscurity, at least for me, because it goes against the name of the property. Which is `receivedAt`. I think if we want to keep the value as the beginning of the round then perhaps property name requires an update.